### PR TITLE
MAT-4415: Layout implementation

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-4415](https://jira.cms.gov/browse/MAT-4415)
(Optional) Related Tickets:

### Summary

This PR 

- Adds an event to be broadcasted from pageHeader to initialize the cql library from CQL library
- Updates dependencies for madie-design-system
- Modifies the Create Measure Dialog to use the reusable dialog from madie-design-system
- Updates related tests

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/PageHeader/PageHeader.tsx [hunk 1]
```diff
@@ -122,7 +121,10 @@ const PageHeader = () => {
               <button
                 className="new-measure"
                 data-testid="create-new-cql-library-button"
-                onClick={() => history.push("/cql-libraries/create")}
+                onClick={() => {
```
Comment (RohitKandimalla) [inline src/components/PageHeader/PageHeader.tsx:124]: className="measures"
className="new-measure"
Can we update these class names and use a generic name? since it is being reused and misleading.

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx [hunk 2]
```diff
@@ -1,153 +1,41 @@
...
-});
 
-const CreateNewMeasureDialog = ({ open, onClose }) => {
+interface Toast {
+  toastOpen: boolean;
+  toastType: string;
+  toastMessage: string;
+}
+
+const CreateNewLibraryDialog = ({ open, onClose }) => {
```
Comment (RohitKandimalla) [inline src/components/NewMeasure/CreateNewMeasureDialog.tsx]: ok, I am a bit confused on this actually works, The component says CreateNewLibraryDialog, and the formik has values related to measures. Also, the file name still says CreateNewMeasureDialog. 
Are we trying to reuse the same dialog for both new object creations? If so who is it being differentiated?

Comment (mcmcphillips) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: Sorry that was an artifact left over from developing. This is the createNewMeasureDialog and it's exported default as such. Just forgot to change it back locally when i was skipping the compilation periods from a locally linked bundle.

We are reusing the same dialog for both dialog parent components

https://github.com/MeasureAuthoringTool/madie-design-system/blob/develop/react/components/MadieDialog/index.jsx

File: src/components/PageHeader/PageHeader.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/components/PageHeader/PageHeader.tsx:124]: Definitely, updated to 'create-button'

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/components/NewMeasure/CreateNewMeasureDialog.tsx]: Also, The logic for the createNewLibrary component is living in madie-cql-library
It listens for a custom event to open and then it has configured logic to close it based on what we pass to the reusable dialog onClose

Right now we have two different approaches to the same problem.
Greg mentioned that we want to have a good separation of concerns, and this prevents us from dumping any more cql-library stuff to layout.

This should probably be a good way to keep everything segmented. Maybe also a different way to do it using util or multiple stores or some variation of models being lazy loaded.

**Review by RohitKandimalla — APPROVED** [review]

**Review by adongare — APPROVED** [review]
